### PR TITLE
Support for specifying handler method inside getHandledMessages

### PR DESCRIPTION
**Pull request by maryo**
Symfony `getHandledMessages` supports specifying a method for each handled message.
https://symfony.com/doc/current/messenger.html#handler-subscriber-options

`__invoke` is the default and is mostly good enough for command buses (one command has one handler). But for event buses it is quite common to have one subscriber listening to multiple events.

OFC method is not the only option and supporting the others would be nice but it was good enough for me.

### CODE REVIEW

File: src/Messenger/DI/MessengerExtension.php [hunk 1]
```diff
@@ -346,7 +351,7 @@ static function (string $serviceName) use ($builder, $busName): bool {
      *
      * @throws InvalidHandlerService
      */
-    private function getHandledMessageNames(string $serviceName): iterable
+    private function getHandledMessages(string $serviceName): iterable
```
Comment (fmasa) [inline src/Messenger/DI/MessengerExtension.php:354]: Can't we handle both types  here and pass a return iterable of DTOs?

Something like:

```php
final class HandlerDefinition
{
    private string $serviceName;
    private string $methodName;
    
    // ...
}
```

This method would then return `iterable<HandlerDefinition>`.

**Comment by fmasa** [comment]
Hi, thank you for opening a PR. Supporting these would be great.
But we should ideally tidy up the current code a bit. Instead of passing `string` everywhere, we should introduce a DTO. Since all classes that need to be touched are `final`, there is no risk of BC break if we change the internal API.

But I am totally on board with adding this capability. :slightly_smiling_face:

**Comment by maryo** [comment]
OK, cool. I've made the DTO but I've used the array index for the message name because otherwise I would need to build the index myself from the iterable. Let me know if that's acceptable or you want more changes. Thanks.

**Comment by fmasa** [comment]
@maryo Thank you for the changes. This is exactly what I had in mind. :slightly_smiling_face: 

Can you please fix errors reported by coding standard check and PHPStan? This is the last thing necessary before merge.

**Comment by maryo** [comment]
@fmasa I forgot to run phpcs locally - fixed. The PHPStan was passing last week. It is caused by this change https://github.com/nette/di/commit/927cf120d0fd1698bd5683dafd0e8c5be4b242f0
which is already reverted in https://github.com/nette/di/commit/475c5261a6da58688e556e1dd7adb0a7ef4edf37
but the commit is not tagged.
I was not able to fix it "properly" so I added the messages into ignored ones. I think PHPStan cannot handle the case with the nullable template parameter with the default value but I am not entirely sure.

**Pull request merged**